core/sql/sqlcomp/CmpSeabaseDDLrepos.cpp

```diff
@@ -358,6 +358,75 @@ short CmpSeabaseDDL::copyOldReposToNew(ExeCliInterface * cliInterface)
   return 0;
 }
 
+short CmpSeabaseDDL::migrateReposViews(ExeCliInterface * cliInterface,
+                                       NABoolean & someViewSaved /* out */)
+{
+  short retcode = 0;  // assume success
+  someViewSaved = FALSE;
+
+  // for each table that has been migrated, save and drop any views
+  // on the old table, and attempt to recreate them on the new
+
+  for (Int32 i = 0; i < sizeof(allReposUpgradeInfo)/sizeof(MDUpgradeInfo); i++)
+    {
+      const MDUpgradeInfo &rti = allReposUpgradeInfo[i];
+
+      if ((! rti.newName) || (! rti.oldName) || (NOT rti.upgradeNeeded))
+        continue;
+
+      Int64 tableUID = getObjectUID(cliInterface,
+        getSystemCatalog(), SEABASE_REPOS_SCHEMA, rti.oldName,
+        COM_BASE_TABLE_OBJECT_LIT, NULL, NULL, FALSE, FALSE /* ignore error */);
+
+      if (tableUID != -1)  // if we got it
+        {
+          NAList<NAString> viewNameList;
+          NAList<NAString> viewDefnList;
+ 
+          short retcode1 = 
+             saveAndDropUsingViews(tableUID, cliInterface,
+                                   viewNameList /* out */,viewDefnList /* out */);
+          if (retcode1)
+            retcode = -1;  // couldn't get views for old repository table
+          else if (viewDefnList.entries() > 0)  // if there are views to migrate
+            {
+              Lng32 firstBadOne = -1;
+              retcode1 = recreateUsingViews(cliInterface, viewNameList, viewDefnList,
+                                            TRUE, &firstBadOne);
+              if (retcode1)
+                {
+                  retcode = -1; 
+                  Lng32 objUID = -2;  // use a fake objectUID since there is no object
+                  
+                  // For any view that could not be migrated, put its definition text
+                  // back in the TEXT table so the user can deal with it later. We
+                  // use a distinctive TEXT_TYPE for this purpose. We have to "make up"
+                  
+                  NABoolean xnWasStartedHere = FALSE;
+                  if (beginXnIfNotInProgress(cliInterface, xnWasStartedHere))
+                    retcode = -1;
+                  else
+                    {                    
+                      for (Lng32 i = firstBadOne; i < viewDefnList.entries(); i++)
+                        {                      
+                          updateTextTable(cliInterface, objUID, COM_BAD_VIEW_TEXT,
+                                          0, viewDefnList[i], NULL, 0, TRUE);
+                          objUID--;  // get another fake object UID
+
+                          someViewSaved = TRUE; 
+                        }
+                      endXnIfStartedHere(cliInterface, xnWasStartedHere, 0);
+                    }                 
+                }                  
+            }
+        }
+      else
+        retcode = -1;  // couldn't get objectUID for old repository table
```

Comment:
Actually, I slept on this overnight and realized that the test at 374 doesn't do the right thing. The way it works now is a view that references a dropped table will just silently be dropped. But better behavior would be to attempt to recreate the view, let that fail, and let the text be written to the TEXT table (as it would for a dropped column, say). I will submit a rework.